feature_tests/src/slices.rs

```diff
@@ -17,6 +17,7 @@ mod ffi {
             Box::new(Self(v.to_string()))
         }
 
+        #[diplomat::attr(supports = memory_sharing, disable)]
```

Comment:
@Manishearth I had an issue with passing owned strings in java as then I needed to pass an arena around which felt contradictory to the idea of an owned parameter.